data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java

```diff
@@ -97,7 +97,9 @@ public ReceiverVerticle(
             final HttpServerOptions httpsServerOptions,
             final Function<Vertx, IngressProducerReconcilableStore> ingressProducerStoreFactory,
             final IngressRequestHandler ingressRequestHandler,
-            final String secretVolumePath) {
+            final String secretVolumePath,
+            final String tlsKeyFilePath, 
+            final String tlsCrtFilePath) {
```

Comment:
You are trying to edit the constructor, but is that necessary to have this? 
```
final String tlsKeyFilePath, 
final String tlsCrtFilePath
```
We can use the way by appending `"tls.key"` and `"tls.crt"` to do so.

Reply:
Can you please explain

Reply:
https://github.com/knative-extensions/eventing-kafka-broker/blob/3fa9eb6db585aa51e45263f8583890a6d418fcfb/data-plane/receiver/src/test/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticleTest.java#L146-L153

As you can see here, in other file, we are creating a new `ReceiverVerticle` object, and we are passing in the `SECRET_VOLUME_PATH` there. 

And in the changes you have made, you are modifying the constructor, which means you will modify all the code that use the constructor of `ReceiverVerticle`. 

So my suggestion here would be: Do not modify the constructor here, we still only pass in `final String secretVolumePath` only. And in the code below when we need to use `tlsKeyFilePath` and `tlsCrtFilePath`, we can just use `secretVolumePath` + "tls.key"` for `tlsKeyFilePath`  and `secretVolumePath` + "tls.crt"` for `tlsCrtFilePath`.

Does that make sense? @MeenuyD

Reply:
should I make this to
```
 final File tlsKeyFile;
 final File tlsCrtFile;
```
this 
```
final String tlsKeyFilePath, 
final String tlsCrtFilePath
```

Reply:
You can get `tlsKeyFilePath` from `secretVolumePath` by doing
```java
this.tlsKeyFile = new File(secretVolumePath + "/tls.key");
```

@MeenuyD

Reply:
Hey @MeenuyD just adding this comment to make it more clear! And apology for not making it clear before.

This is the constructor of `ReceiverVerticle`, and you don't make any changes here, it stays the same.

https://github.com/knative-extensions/eventing-kafka-broker/blob/e7017efef281a046280e27da77d05101632b3d53/data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java#L94-L101


This is where you create the variable within the class
https://github.com/knative-extensions/eventing-kafka-broker/blob/e7017efef281a046280e27da77d05101632b3d53/data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java#L115-L116

Taking `tlsKey` file as an example:
Here instead of having `this.tlsKeyFilePath`, you replace it with a new variable `this.tlsKeyFile` by doing this 
```
this.tlsKeyFile = new File(secretVolumePath + "/tls.key");
```

Same apply to `tlsCrt`!

Then after you make those changes, you can directly use those new variables here.

https://github.com/knative-extensions/eventing-kafka-broker/blob/e7017efef281a046280e27da77d05101632b3d53/data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java#L134-L148

Instead of
```
if (tlsKeyFile.exists() && tlsCrtFile.exists() && httpsServerOptions != null)
```
you can do
```
if (this.tlsKeyFile.exists() && this.tlsCrtFile.exists() && httpsServerOptions != null)
```

Reply:
Thank you for your guidance can you please review it